addons/misra.py

```diff
@@ -4093,14 +4124,17 @@ def setSuppressionList(self, suppressionlist):
 
                 self.addSuppressedRule(ruleNum)
 
-    def reportError(self, location, num1, num2):
+    def reportError(self, location, num1, num2,otherID = None):
```

Comment:
please add space before otherID

I would like that code in misra.py is "pythonic" overall. The exception is when we want that the name of a function/class/etc matches the corresponding name in Cppcheck.

So in this case I would suggest the name `other_id`.

You are the python expert here so well if you see more things that can be more pythonic then feel free to suggest it..